ostd/src/logger.rs

```diff
@@ -2,36 +2,60 @@
 
 //! Logging support.
 
+use alloc::format;
+
 use log::{Level, Metadata, Record};
+use owo_colors::OwoColorize;
 
-use crate::early_println;
+use crate::{arch::timer::Jiffies, boot::kernel_cmdline, early_println};
 
 const LOGGER: Logger = Logger {};
 
-/// The log level.
-///
-/// FIXME: The logs should be able to be read from files in the userspace,
-/// and the log level should be configurable.
-pub const INIT_LOG_LEVEL: Level = Level::Error;
+/// The log level. It will change during `logger::init`
+static mut LOG_LEVEL: Level = Level::Error;
 
 struct Logger {}
 
 impl log::Log for Logger {
     fn enabled(&self, metadata: &Metadata) -> bool {
-        metadata.level() <= INIT_LOG_LEVEL
+        // SAFETY: Once changed in `logger::init`, LOG_LEVEL is guaranteed to be immutable during
+        // the system lifetime.
+        unsafe { metadata.level() <= LOG_LEVEL }
```

Comment:
Only enclose the code that needs unsafe. Here the code is `LOG_LEVEL`, not `metadata.level()`.